docs/getting-started/config-examples/CustomWebserviceDataObject-1.scala

```diff
@@ -0,0 +1,114 @@
+ */
+case class CustomWebserviceDataObject(override val id: DataObjectId,
+                                      override val metadata: Option[DataObjectMetadata] = None,
+                                      additionalHeaders: Map[String,String] = Map(),
+                                      queryParameters: Option[Seq[DepartureQueryParameters]] = None,
+                                      timeouts: Option[HttpTimeoutConfig] = None,
+                                      authMode: Option[AuthMode] = None,
+                                      baseUrl : String,
+                                      nRetry: Int = 1)
+                                     (@transient implicit val instanceRegistry: InstanceRegistry)
+  extends DataObject with CanCreateDataFrame with SmartDataLakeLogger {
+
+  private var dfCached : Option[DataFrame] = None
+
+
+  private val now = Instant.now.getEpochSecond
+  // check whether there are query parameters available from the config
+  if(queryParameters == None){
+    throw new ConfigurationException(s"($id) no query parameters available")
+  }
+
+  // if we have query parameters in the state we will use them from now on
+  val currentQueryParameters = queryParameters.get
+
+  @tailrec
+  private def request(url: String, method: WebserviceMethod = WebserviceMethod.Get, body: String = "", retry: Int = nRetry) : Array[Byte] = {
+    val webserviceClient = ScalaJCustomWebserviceClient(this, Some(url))
+    val webserviceResult = method match {
+      case WebserviceMethod.Get => webserviceClient.get()
+      case WebserviceMethod.Post => webserviceClient.post(body.getBytes, "application/json")
+    }
+    webserviceResult match {
+      case Success(c) =>
+        logger.info(s"Success for request ${url}")
+        c
+      case Failure(e) =>
+        if(retry == 0) {
+          logger.error(e.getMessage, e)
+          throw new WebserviceException(e.getMessage)
+        }
+        logger.info(s"Request will be repeated, because the server responded with: ${e.getMessage}. \nRequest retries left: ${retry-1}")
+        request(url, method, body, retry-1)
+    }
+  }
+
+  override def getDataFrame(partitionValues: Seq[PartitionValues])(implicit session: SparkSession, context: ActionPipelineContext): DataFrame = {
+    import org.apache.spark.sql.functions._
+    implicit val formats: Formats = DefaultFormats
+    import session.implicits._
+
+    if(dfCached.isEmpty) {
+      // given the query parameters, generate all requests
+      val departureRequests = currentQueryParameters.map(
+        param => s"${baseUrl}?airport=${param.airport}&begin=${param.begin}&end=${param.end}"
+      )
+
+      // make request
+      val departuresResponses = departureRequests.map(request(_))
+      // deserialize the result into a sequence of Departure objects
+      val departures = departuresResponses.flatMap(res => JsonMethods.parse(new String(res)).extract[Seq[Departure]])
```

Comment:
Thanks for the review. You are right, it might be a bit confusing, but both choices have been done on purpose. The idea is to show two ways of parsing the json. The initial version uses json4s and then the final version, which is implemented by the user, uses from_json. Concerning the schema, yes, that could be an configuration parameter. Here, I wanted to show that we can use the different phases of SDL to define the schema in the Init phase and then to make the request in the Exec phase. It might be that the schema can be grabbed from an additional endpoint, this would be done in Init phase. Therefore I chose this design.

Reply:
I agree with @hiltpold. But maybe i'm biaised because I went through the previous version of the tutorial, had the remark about "why do we download data in the init phase and do nothing in the exec phase" and now I find my comment addressed :P 

I agree that ideally the schema should be in the config. But I don t know how much more overhead this is to explain in the tutorial, so I would be fine with it in the code at this stage. Maybe we can mention that ideally it should be in a config file?
So we got one reviewer confused, and another one reviewer happy. @pgruetter ?

Reply:
Sorry for not adding to this discussion, think it's solved now after offline discussions. The schema is now in the configuration file, which I think is important. And we still show and explain, how to distinguish between Init and Exec phase in a custom implementation.